Mathlib/Algebra/Ring/CentroidHom.lean

```diff
@@ -467,21 +467,44 @@ lemma centerToCentroid_apply (z : { x // x ∈ NonUnitalSubsemiring.center α })
     (centerToCentroid z) a = z * a := rfl
 
 lemma center_iff_op_centroid (a : α) :
-    a ∈ NonUnitalSubsemiring.center α ↔ L a = R a ∧ (L a) ∈ Set.range CentroidHom.toEnd := by
+    a ∈ NonUnitalSubsemiring.center α ↔ R a = L a ∧ (L a) ∈ RingHom.rangeS (toEndRingHom α) := by
   constructor
-  · exact fun ha ↦ ⟨AddMonoidHom.ext <| IsMulCentral.comm ha, ⟨centerToCentroid ⟨a, ha⟩, rfl⟩⟩
+  · exact fun ha ↦ ⟨AddMonoidHom.ext <| fun _ => (IsMulCentral.comm ha _).symm,
+      ⟨centerToCentroid ⟨a, ha⟩, rfl⟩⟩
   · rintro ⟨hc, ⟨T, hT⟩⟩
     have e1 (d : α) : T d = a * d := congr($hT d)
-    have e2 (d : α) : T d = d * a := congr($(hT.trans hc) d)
+    have e2 (d : α) : T d = d * a := congr($(hT.trans hc.symm) d)
     constructor
-    case comm => exact (congr($hc ·))
+    case comm => exact (congr($hc.symm ·))
     case left_assoc => simpa [e1] using (map_mul_right T · ·)
     case mid_assoc => exact fun b c ↦ by simpa [e1 c, e2 b] using
       (map_mul_right T b c).symm.trans <| map_mul_left T b c
     case right_assoc => simpa [e2] using (map_mul_left T · ·)
 
 end NonUnitalNonAssocSemiring
 
+section NonUnitalNonAssocCommSemiring
+
+variable [NonUnitalNonAssocCommSemiring α]
+
+local notation "L" => AddMonoid.End.mulLeft
+local notation "R" => AddMonoid.End.mulRight
+
+lemma center_iff_op_mul_commute (a : α) :
+    a ∈ NonUnitalSubsemiring.center α ↔ ∀ b : α, Commute (L b) (L a) := by
```

Comment:
```suggestion
lemma center_iff_op_mul_commute (a : α) :
    a ∈ NonUnitalSubsemiring.center α ↔ ∀ b : α, Commute (L b) (L a) := by
```
This seems to highlight an asymmetry which doesn't actually exist. Can you provide a lemma of the form `Commute (L a) (L b) ↔ Commute (R a) (R b)` (apologies if it exists already and I missed it).

Reply:
`AddMonoid.End.comm_mulRight_eq_mulLeft` introduced in https://github.com/leanprover-community/mathlib4/pull/7060/files#diff-f49b5ab38f9aab47a03faa24fb627f8c9c57f2175af4c3c4bf7f186ebdbb3673R360 is the result that the left and right multiplication operators coincide on Commutative Semirings. Following Hanche-Olsen and Størmer, I've changed the notation for `AddMonoid.End.mulLeft` from `L` to `T` in this case to de-emphasise the "leftness". Is that sufficient?

Reply:
Aha, I missed commutativity. I think you should leave it as `L`, but add a docstring emphasizing that `mulLeft` and `mulRight` coincide because the ring is commutative. This could be called `mem_center_iff_commute` or, again because the namespace `CentroidHom` is a bit weird: `NonUnitalNonAssocCommSemiring.mem_center_iff`.

Reply:
Thanks, I've added a docstring and renamed the lemma.